OpenRA.Game/Renderer.cs

```diff
@@ -85,15 +85,15 @@ public Renderer(IPlatform platform, GraphicSettings graphicSettings)
 			TempBufferSize = graphicSettings.BatchSize;
 			SheetSize = graphicSettings.SheetSize;
 
-			WorldSpriteRenderer = new SpriteRenderer(this, Context.CreateShader("combined"));
+			WorldSpriteRenderer = new SpriteRenderer(this, Context.CreateUnsharedShader<CombinedShaderBindings>());
 			WorldRgbaSpriteRenderer = new RgbaSpriteRenderer(WorldSpriteRenderer);
 			WorldRgbaColorRenderer = new RgbaColorRenderer(WorldSpriteRenderer);
-			WorldModelRenderer = new ModelRenderer(this, Context.CreateShader("model"));
-			SpriteRenderer = new SpriteRenderer(this, Context.CreateShader("combined"));
+			WorldModelRenderer = new ModelRenderer(this);
```

Comment:
It is probably feasible to implement the other `World*Renderer`s using the same plumbing, so we could perhaps use `IWorldRenderer` instead of `ICustomRenderer`? This might cause confusion against the actual `WorldRenderer` class though.